openedx/core/djangoapps/content/course_metadata/tasks.py

```diff
@@ -0,0 +1,33 @@
+"""
+This module has implementation of celeary tasks to compute course aggregate metadata
+"""
+
+import logging
+
+from celery.task import task
+from opaque_keys.edx.keys import CourseKey
+from .models import CourseAggregatedMetaData
+from api_manager.courseware_access import get_course_leaf_nodes
```

Comment:
I think we need to break this dependency -- we don't want other apps in the system to depend on the orchestration layer -- we should stick to top-down as much as possible, even if it means relocating the get_course_leaf_nodes operation.  The good news is the only dependency on this operation in api_manager has been removed with this change set, so it could be relocated to the new app.